cli/src/commands/fix.rs

```diff
@@ -195,20 +194,41 @@ async fn fix_one_file(
             .read_file(&file_to_fix.repo_path, &file_to_fix.file_id)
             .await?;
         read.read_to_end(&mut old_content).await?;
-        let new_content = matching_tools.fold(old_content.clone(), |prev_content, tool_config| {
-            match run_tool(
-                workspace_root,
-                &tool_config.command,
-                file_to_fix,
-                &prev_content,
-            ) {
-                Ok(next_content) => next_content,
-                // TODO: Because the stderr is passed through, this isn't always failing
-                // silently, but it should do something better will the exit code, tool
-                // name, etc.
-                Err(_) => prev_content,
-            }
-        });
+        let new_content =
+            matching_tools.fold(
+                old_content.clone(),
+                |prev_content, tool_config| match run_tool(
+                    workspace_root,
+                    &tool_config.command,
+                    file_to_fix,
+                    &prev_content,
+                ) {
+                    Ok(next_content) => next_content,
+                    Err(Some(e)) => {
+                        let stderr_text = if e.stderr.is_empty() {
+                            "(no stderr)".to_string()
+                        } else {
+                            format!("'{}'", String::from_utf8_lossy(&e.stderr).trim())
+                        };
+
+                        writeln!(
+                            ui.warning_default(),
+                            "tool '{}' exited with status {} when fixing '{}': {}",
+                            tool_config.name,
+                            e.status.code().unwrap_or_default(),
+                            // We use `RepoPath::as_internal_file_string` in this warning
+                            // as it's what is (optionally) passed as an argument to the
+                            // tool.
```

Comment:
Hmm, that seems like a minor bug. @hooper, do you remember if there's a reason we don't format the path using using backslashes on Windows (as `path.to_fs_path()` would do)?